### PR TITLE
Split visit command into home and visit

### PR DESCRIPTION
**Pull request by SirYwell**
This PR introduces a new command `home` to make the visit command less ambiguous.

### CODE REVIEW

**Comment by RedstoneFuture** [comment]
```
usage = "/plot visit [<player>|<alias>|<world>|<id>] [#]",
usage = "/plot home [<page>|<alias>|<area;x;y>|<area> <x;y>]",
```
Good, with this split we can support number-playernames like [2234](https://de.namemc.com/profile/2234.1), I think.


Can two different people have the same alias for 2 different plos? Or is "/plot home <alias>" the same as "/plot visit alias"?

How com with "/plot visit world 1"? To the first plot of "world"?

What do you think about:

```
usage = "/plot visit [<player> <#>|<alias>|<world or area> <x;y>]",
usage = "/plot home [<#>]",
```

**Comment by SirYwell** [comment]
> Good, with this split we can support number-playernames
That was the main reason for this change.

Aliases are unique per plot area, so `/plot gome <alias>` and `/plot visit <alias>` will target the same plot, although home only allows you to teleport to plots you're the owner of, whereas you can teleport to other's plot using visit (if the required permissions are set).

Visit is meant to visit plots of other players, using either their name, their name and a number, a plot id or an alias.

File: Core/src/main/java/com/plotsquared/core/command/Visit.java [hunk 1]
```diff
@@ -54,8 +54,8 @@
 @CommandDeclaration(command = "visit",
     permission = "plots.visit",
     description = "Visit someones plot",
-    usage = "/plot visit [<player>|<alias>|<world>|<id>] [#]",
-    aliases = {"v", "tp", "teleport", "goto", "home", "h", "warp"},
+    usage = "/plot visit <<player>|<alias>|<world>> [#]",
```
Comment (Sauilitired) [inline Core/src/main/java/com/plotsquared/core/command/Visit.java:57]: This is no longer correct, is it? Should probably be `<player>|<alias>|<plot>`

File: Core/src/main/java/com/plotsquared/core/command/Visit.java [hunk 2]
```diff
@@ -155,7 +155,7 @@ private void visit(@NotNull final PlotPlayer player, @NotNull final PlotQuery qu
         switch (args.length) {
             // /p v [...] [...] <page>
             case 3:
-                if (!MathMan.isInteger(args[1])) {
+                if (!MathMan.isInteger(args[2])) {
```
Comment (Sauilitired) [inline Core/src/main/java/com/plotsquared/core/command/Visit.java:158]: Is this then `/plot visit <user> <area> <number>`?

File: Core/src/main/java/com/plotsquared/core/command/Visit.java [hunk 3]
```diff
@@ -214,61 +203,31 @@ private void visit(@NotNull final PlotPlayer player, @NotNull final PlotQuery qu
...
+                Captions.COMMAND_SYNTAX.send(player, getUsage());
+                return CompletableFuture.completedFuture(false);
             default:
         }
 
         return CompletableFuture.completedFuture(true);
     }
 
-    public Collection<Command> tab(PlotPlayer player, String[] args, boolean space) {
+    @Override public Collection<Command> tab(PlotPlayer player, String[] args, boolean space) {
```
Comment (Sauilitired) [inline Core/src/main/java/com/plotsquared/core/command/Visit.java:230]: in `case 0` it still completes numbers, this should only be done for `/plot home`. `case 1` is also wrong as it assumes a number for the first argument, which is only the case for home, so the entire tab completion here is wonky

File: Core/src/main/java/com/plotsquared/core/command/HomeCommand.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+                query.withPlot(plot);
+                break;
+            case 0:
+                query.withSortingStrategy(SortingStrategy.SORT_BY_CREATION);
+                break;
+        }
+        home(player, query, page, confirm, whenDone);
+        return CompletableFuture.completedFuture(true);
+    }
+
```
Comment (Sauilitired) [inline Core/src/main/java/com/plotsquared/core/command/HomeCommand.java:168]: missing completions?

File: Core/src/main/java/com/plotsquared/core/command/Visit.java (hunk 2 above)
Comment (SirYwell) [reply Core/src/main/java/com/plotsquared/core/command/Visit.java:158]: Yes, seems to be the only remaining part with 3 arguments. So the usage should be `/plot visit <player>|<alias>|<plot> [area] [#]` I guess?

Comment (Sauilitired) [reply Core/src/main/java/com/plotsquared/core/command/Visit.java:158]: it's a bit messier than that, `/plot visit <player>|<alias>|<plot> [area]|[#] [#]` lol

**Comment by RedstoneFuture** [comment]
`usage = "/plot home [<page>|<alias>|<area;x;y>|<area> <x;y>|<area> <page>]",` Ok.

`usage = "/plot visit <player>|<alias>|<plot> [area]|[#] [#]",`

Do you mean:

`usage = "/plot visit <<player>|<alias>|<area;x;y>|<area> <x;y>|<area>> [page]",`

/p v testplayer
/p v testplayer 2
/p v testalias
/p v world;1;1
/p v world 1;2
/p v world 2

**Comment by RedstoneFuture** [comment]
**Tests | worked commands:**

/p h
/p h <#>
/p h <alias> **(Currently not with merged plots!)**
/p h <X>;<Y>
/p h <area/world> <X>;<Y>
/p h <area/world>;<X>;<Y>
/p h <area/world> <#>


/p v <player>
/p v <player> <area/world>
/p v <player> <area/world> <#>
/p v <alias>
/p v <X>;<Y>
/p v <area/world>;<X>;<Y>


**`plots.visit.other` still needed. Maybe we remove this permission check.**
/p v <area/world> <X>;<Y> **not working (consistence ? Or unimpossible? )**

**Pull request merged**